src/lib.rs

```diff
@@ -653,7 +678,11 @@ impl Layer {
             ],
             TiledError::MalformedAttributes("layer must have a name".to_string())
         );
-        let mut tiles = Vec::new();
+        let mut tiles = TilesContainer {
```

Comment:
Could this be initialized with correct data from the start? I feel like this could lead to invalid states where the tiles member never got initialized.